modules/tensor_mechanics/include/materials/ADRankTwoCylindricalComponent.h

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADMaterial.h"
+#include "RankTwoTensor.h"
+
+/**
+ * MaterialRankTwoCylindricalComponent uses the namespace RankTwoScalarTools to compute scalar
+ * values from Rank-2 tensors.
+ */
+class ADRankTwoCylindricalComponent : public ADMaterial
+{
+public:
+  static InputParameters validParams();
+
+  ADRankTwoCylindricalComponent(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void computeQpProperties() override;
+
+  const ADMaterialProperty<RankTwoTensor> & _tensor;
+
+  /// Name of the stress/strain to be calculated
+  const std::string _property_name;
+
+  /// Stress/strain value returned from calculation
+  ADMaterialProperty<Real> & _property;
+
+  /**
```

Comment:
Use the single line `///` comments for describing variables, even if it takes a couple of lines.